### ISSUE TITLE
--allow-hosts allowing more than specified single host

### ISSUE THREAD

**Issue by Joshfindit** [mitmproxy/mitmproxy#5156]
#### Problem Description
While using `mitmproxy` to observe traffic to a single domain, the UI is cluttered with other traffic.

The actual command I'm running:
`mitmproxy --ssl-insecure --allow-hosts 'discord.gift'`

I'm then configuring a single iOS device to use the proxy.

Instead of that single domain, I see traffic to non-SSL services on the LAN.
I (now) understand that the subtext of `--allow-hosts` is "allow traffic to these hosts to be decrypted/MITMd", but that is not clear from the command. I feel like this is a bug because the documentation suggests that `--allow-hosts` allows us to focus only on traffic to the host(s) specified and it does not make any mention of HTTP vs HTTPS.

#### Steps to reproduce the behavior:
1. Run `mitmproxy` with `--allow-hosts 'example.com'`
2. Pass HTTP traffic through `mitmproxy`
3. View the requests inside `mitmproxy`

#### System Information
Paste the output of "mitmproxy --version" here.
```
Mitmproxy: 7.0.4 binary
Python:    3.9.7
OpenSSL:   OpenSSL 1.1.1l  24 Aug 2021
Platform:  Linux-5.4.162-1-pve-x86_64-with-glibc2.27
```

**Comment by mhils** [comment]
To confirm, your point is that plain HTTP traffic to other hosts should not appear in the UI if `--allow-hosts` is passed?

https://docs.mitmproxy.org/dev/howto-ignoredomains/#limitations and https://docs.mitmproxy.org/dev/howto-ignoredomains/#fn:1 in particular should explain why that is the case. I would be more than happy to merge a PR that adds a hint in the docs that you can filter the view instead (press `f` in mitmproxy, or set the `view_filter` option).

**Comment by Joshfindit** [comment]
Yes you've got it. My intuition says "--allow-hosts is a whitelist, and therefore I should only see traffic to/from that host" and so HTTP requests showing up violates that intuition.

I read the docs and understand that this is a limitation that `mitmproxy` is working with. I'm going to work with this issue a bit and see if I can get to a solid way to do what I'm looking for before seeing if I have enough to submit a documentation PR.

**Comment by Joshfindit** [comment]
@mhils Can you confirm that the closest users can currently get is the following:

`mitmproxy --allow-hosts 'discord.gift' --view-filter '~d discord.gift'`

And that it sets `mitmproxy` to:
1. Intercept HTTPS requests only to `discord.gift`
2. Intercept all HTTP requests, but in the UI only show requests to/from the domain `discord.gift`

And that the final effect to the user is an interface that shows HTTP and HTTPS requests to/from that domain without interfering with HTTPS requests to any other domain.

**Comment by mhils** [comment]
Yes, that is correct! Sorry for the slow reply. :)

**Comment by otheus** [comment]
@mhils Just to clarify the summary by @Joshfindit, mitmproxy still acts on HTTPS requests other than _allow-hosts_ by proxying the connections as a normal proxy service. 

I contend that any requests not included in _allow-hosts_ should still be displayed by default, unless excluded by the _view-filter_. However, the status/info portion should indicate the connection is being passed through.

**Comment by mhils** [comment]
> @mhils Just to clarify the summary by @Joshfindit, mitmproxy still acts on HTTPS requests other than _allow-hosts_ by proxying the connections as a normal proxy service.

Yes, correct. More specifically, TLS connections are not intercepted.

> I contend that any requests not included in _allow-hosts_ should still be displayed by default, unless excluded by the _view-filter_. However, the status/info portion should indicate the connection is being passed through.

`allow_hosts` just has a terrible name, I plan to eventually rename it to something akin to `ignore_hosts_except`. At which point the current behavior makes more sense again.

**Comment by Joshfindit** [comment]
> `allow_hosts` just has a terrible name, I plan to eventually rename it to something akin to `ignore_hosts_except`. At which point the current behavior makes more sense again.

Since HTTP requests are still shown, I would suggest going even further with `ignore_https_hosts_except`

**Comment by mhils** [comment]
Yeah, I see where you're coming from. The waters get a bit murky because it also applies to plain TCP proxying, it's only plain HTTP without CONNECT that we cannot filter. So maybe it makes sense to add a bunch of boilerplate to filter out that specific case as well. :)

**Comment by otheus** [comment]
@Joshfindit I had an idea that is more precise version of these options: https://github.com/mitmproxy/mitmproxy/issues/5270
